src/input.c

```diff
@@ -38,3 +40,228 @@ char* ReadUserInput(const char* const message) {
+               max_value);
+      }
+    }
+    free(attribute);
+  }
+
+  return output;
+}
+
+bool GetInputBoolean(const char* message) {
+  while (1) {
+    char* attribute_bool = ReadUserInput(message);
+
+    char attribute_bool_answer = attribute_bool[0];
+    free(attribute_bool);
+
+    switch (attribute_bool_answer) {
+      case kYes:
+        return true;
+
+      case kNo:
+        return false;
+
+      default:
+        printf("Invalid answer!\n");
+        break;
+    }
+  }
+}
+
+UserPreferences ManualAttributes(void) {
+  UserPreferences attributes;
+  const int kMaxNum = 10;
+  const int kMaxCarEfficiency = 500;
+  const double kConvertToAttributeDouble = 10.0;
+
+  printf(
+      "To input your personal preferences please write a number between "
+      "0-10!\n");
+
+  double attribute_price =
+      GetInputInteger("Price preference (0-10):", kMaxNum) /
+      kConvertToAttributeDouble;
+  double attribute_time = GetInputInteger("Time preference (0-10):", kMaxNum) /
+                          kConvertToAttributeDouble;
+  double attribute_environment =
+      GetInputInteger("Sustainability preference (0-10):", kMaxNum) /
+      kConvertToAttributeDouble;
+  double attribute_health =
+      GetInputInteger("Health preference (0-10):", kMaxNum) /
+      kConvertToAttributeDouble;
+
+  double attribute_total = attribute_price + attribute_health + attribute_time +
+                           attribute_environment;
+
+  // Normalize the attributes to between 0 and 1.
+  attribute_price = attribute_price / attribute_total;
+  attribute_time = attribute_time / attribute_total;
+  attribute_environment = attribute_environment / attribute_total;
+  attribute_health = attribute_health / attribute_total;
+
+  bool attribute_walk =
+      GetInputBoolean("Can your route include walking? (y/n):");
+  bool attribute_bike =
+      GetInputBoolean("Can your route include biking? (y/n):");
+  bool attribute_train =
+      GetInputBoolean("Can your route include trains? (y/n):");
+  bool attribute_car = GetInputBoolean("Can your route include a car? (y/n):");
+
+  // Transfer the user data to the UserPreferences struct.
+  attributes.price = attribute_price;
+  attributes.time = attribute_time;
+  attributes.environment = attribute_environment;
+  attributes.health = attribute_health;
+  // Save attributes to file.
+  SetUserPreference("price", attribute_price);
+  SetUserPreference("time", attribute_time);
+  SetUserPreference("environment", attribute_environment);
+  SetUserPreference("health", attribute_health);
+
+  attributes.walk = attribute_walk;
+  attributes.bike = attribute_bike;
+  attributes.train = attribute_train;
+
+  if (attribute_car) {
+    int attribute_car_efficiency = GetInputInteger(
+        "What is the efficiency of the car in (UNIT):", kMaxCarEfficiency);
+    attributes.car = attribute_car;
+    attributes.car_fuel_efficiency = attribute_car_efficiency;
+  } else {
+    attributes.car = false;
```

Comment:
This line may be redundant, as attributes.car is already false if it is not true in line 142

Reply:
Will change